### PR TITLE
Assigment pandas EloiAncellin

### PR DESCRIPTION
**Pull request by EloiAncellin**

### CODE REVIEW

**Comment by agramfort** [comment]
good job @EloiAncellin 

can you just remove all the unecessary files ? you should not need to add any new file in this PR

**Comment by EloiAncellin** [comment]
Bonjour Monsieur,

C’est fait !

Excellent soirée,

Eloi

File: .idea/.gitignore [hunk 1]
```diff
@@ -0,0 +1,2 @@
+# Default ignored files
+/workspace.xml
```
Comment (agramfort) [inline .idea/.gitignore:2]: please remove all  the .idea folder

Comment (agramfort) [reply .idea/.gitignore:2]: I see 10 files modified. I should see only one

Comment (EloiAncellin) [reply .idea/.gitignore:2]: I have made an mistake and deleted the test file, thus, I am correcting my errors...

**Comment by agramfort** [comment]
you should not change the data/referendum.csv file

**Comment by EloiAncellin** [comment]
Did I do it ?

**Comment by agramfort** [comment]
see the diff in your PR

https://github.com/x-datascience-datacamp/datacamp-assignment-pandas/pull/10/files

I see 3 files changed

**Comment by tomMoral** [comment]
If you are finished, you can sign up for this RAMP challenge: https://ramp.studio/events/brevet_datacamp2020
This is what we are going to do this afternoon. The goal is to try and solve the challenge.
You need to click on the blue button: `Join the event` and I need to approve your participation so send me a message once you have done this.
You can then read the starting kit and go to the left menu to submit your solutions in the sandbox.

File: pandas_questions.py [hunk 2]
```diff
@@ -28,8 +27,12 @@ def merge_regions_and_departments(regions, departments):
     The columns in the final DataFrame should be:
     ['code_reg', 'name_reg', 'code_dep', 'name_dep']
     """
-
-    return pd.DataFrame({})
+    return pd.merge(regions, departments, left_on='code', right_on='region_code', how='left'
+                    ).drop(
+        ['code_x', 'id_x', 'slug_x', 'slug_y', 'id_y'], axis=1
+    ).rename(columns=
+             {'region_code': 'code_reg', 'name_x': 'name_reg', 'code_y': 'code_dep', 'name_y': 'name_dep'}
```
Comment (tomMoral) [inline pandas_questions.py:34]: Not pep8.

File: pandas_questions.py [hunk 3]
```diff
@@ -61,12 +73,22 @@ def plot_referendum_map(referendum_result_by_regions):
       should display the rate of 'Choice A' over all expressed ballots.
     * Return a gpd.GeoDataFrame with a column 'ratio' containing the results.
     """
+    geo_reg = gpd.read_file('./data/regions.geojson')
 
-    return gpd.GeoDataFrame({})
+    geo_referundum = pd.merge(geo_reg, referendum_result_by_regions, left_on='code', right_on='code_reg', how='inner')
+    geo_referundum['ratio'] = geo_referundum['Choice A'] / (
+            geo_referundum['Registered'] - geo_referundum['Abstentions'] - geo_referundum['Null'])
```
Comment (tomMoral) [inline pandas_questions.py:80]: pep8

**Review by tomMoral — CHANGES_REQUESTED** [review]
Nice one!

Just fix pep8.

**Pull request closed without merging**